### DISCUSSION TITLE
有了 Prisma 就别用 TypeORM 了 | 愧怍的小站

### DISCUSSION THREAD

**Discussion by giscus** [kuizuo/blog#266] [General]
# 有了 Prisma 就别用 TypeORM 了 | 愧怍的小站

要说 2024 年 Node.js 的 ORM 框架应该选择哪个？毫无疑问选 Prisma。至于为何，请听我细细道来。

https://kuizuo.cn/blog/with-prisma-dont-use-typeorm/

**Comment by MarvinXu** [comment]
写的真不错！

**Comment by MarvinXu** [comment]
zod 和 class-validator 推荐哪个啊

**Reply by kuizuo** [reply]
zod, 我会在过几天开源一个nest + trpc+ zod +  prisma的一个组合，到时候你可以借鉴一下。

**Comment by songispm** [comment]
对比 drizzle-orm 呢，大佬也给讲讲呗

**Reply by kuizuo** [reply]
因为我是 Typeorm 和 Primsa 的重度使用用户了，因此在我开发体验中，能不选 Typeorm 就别选，也才有感而发创作这篇文章。

至于说 drizzle-orm 我没怎么使用过，但从 prisma 和 drizzle-orm 的背后是否有商业公司在推动能决定这个工具库是否能有维护的保证，而 Prisma 背后就是，你不必担心它会黄掉或者不火(概率很小)。

**Reply by yisibl** [reply]
可以看看这两篇文章：

- https://betterstack.com/community/guides/scaling-nodejs/drizzle-vs-prisma/
- https://blog.rxliuli.com/p/4140cb8be8b044a5bb9cebe0930e3a4a/

**Comment by cvpv** [comment]
嗨！我正在基于midway开发，我注意到官方推荐的就是TypeORM，Sequelize、MikroORM，您觉得其中哪个ORM框架比较好呢？

**Comment by 1161853339** [comment]
Prisma 两个恶心的坑，第一个如果参数是 undefined，他妈的能把数据全部查出来，还有一个坑是 datetime 在 sqlite上简直玄乎，都不知道怎么样才合适，后面直接用int存时间戳

**Comment by zamlia** [comment]
getOne查询实体外的聚合数据当时给我研究了好久，结果发现getRawOne才行，而且实体字段还会有前缀，直接无解
